Tests/BaseTest.php

```diff
@@ -1,16 +1,83 @@
 <?php
 
-namespace Cybex\Protector\Tests;
-
 use Cybex\Protector\ProtectorServiceProvider;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Storage;
 use Orchestra\Testbench\TestCase;
 
 abstract class BaseTest extends TestCase
 {
+    /**
+     * @return string
+     */
+    protected function getBasePath(): string
+    {
+        return __DIR__ . '/scaffolds/includes-protector';
+    }
+
+    /**
+     * @param $app
+     * @return string[]
+     */
     protected function getPackageProviders($app): array
     {
         return [
             ProtectorServiceProvider::class,
         ];
     }
+
+    /**
+     * @param $method
+     *
+     * @return ReflectionMethod
+     */
+    protected function getAccessibleReflectionMethod($method): ReflectionMethod
+    {
+        $reflectionProtector = new ReflectionClass($this->protector);
+        $method              = $reflectionProtector->getMethod($method);
+
+        $method->setAccessible(true);
+
+        return $method;
+    }
+
+    /**
+     * Allows a test to call a protected method.
+     *
+     * @param string $methodName
+     * @param array $params
+     * @return mixed
+     */
+    protected function runProtectedMethod(string $methodName, array $params): mixed
+    {
+        $method = $this->getAccessibleReflectionMethod($methodName);
+        return $method->invoke($this->protector, ...$params);
+    }
+
+    /**
+     * @param $fileNames
+     * @return void
+     */
+    protected function provideDynamicDumps($fileNames): void
```

Comment:
whats a dynamic dump

Reply:
There is no dynamic dump. But in some places in the test we need to provide a dynamic number of different kinds of dumbs to cover up the different test cases. Though, I will think about a better name.

Reply:
Method will be renamed in bugfix/diskusage.